packages/application-generic/src/factories/chat/handlers/base.handler.ts

```diff
@@ -0,0 +1,26 @@
+import { IChatOptions, IChatProvider } from '@novu/stateless';
+import { ChannelTypeEnum } from '@novu/shared';
+import { IChatHandler } from '../interfaces';
+
+export abstract class BaseChatHandler implements IChatHandler {
+  protected provider: IChatProvider;
+
+  protected constructor(
+    private providerId: string,
+    private channelType: string
+  ) {}
+
+  canHandle(providerId: string, channelType: ChannelTypeEnum) {
+    return providerId === this.providerId && channelType === this.channelType;
+  }
+
+  abstract buildProvider(credentials);
+
+  async send(chatContent: IChatOptions) {
+    if (process.env.NODE_ENV === 'test') {
```

Comment:
A potential improvement would be isolating these checks in a helper method inside of the package that we could rely onto. E.g.:
```
export const isTestEnvironment = () => process.env.NODE_ENV === 'test'
```

Reply:
So this is just code I have moved from api to application generic and would not like to do a lot of changes in it before it is merged...